.github/workflows/test.yml

```
@@ -0,0 +1,27 @@
+name: Ruby CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        ruby-version: ['3.2', '3.1', '3.0']
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Ruby ${{ matrix.ruby-version }}
+        uses: ruby/setup-ruby@ec02537da5712d66d4d50a0f33b7eb52773b5ed1
```

```suggestion
        uses: ruby/setup-ruby@v1
```

でよかったりしません？コミットで指定したいこだわりがあったりするのかな。

---

https://docs.github.com/ja/actions/automating-builds-and-tests/building-and-testing-ruby
> GitHub では、コミット SHA にアクションをピン留めすることが推奨されます。
> 新しいバージョンを取得するには、SHA を更新する必要があります。
> タグまたはブランチを参照することもできますが、アクションは警告なしに変更される可能性があります。

公式にはコミットハッシュ固定を推奨しています
が、固定してるバージョンに脆弱性があったときのリスクなどを考慮すると`@v1`で指定しておいて壊れたら都度直すのとどっちが良いのか

---

公式もそう書かれてるのか。 BP なのは知っていたんですが。
その通りで、 ruby org で管理してる setup-ruby ならバージョン指定で大丈夫そう、というコメントでした。一方でよくわからんところのもので信用できないと判断したらコミットでいいかなくらいで生きていましたね。
コミット指定、安全ではありますが面倒見るのが大変なので...

---

(Approve しているのでこれは変えてもいいし変えなくてもいいです)